keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,1114 @@
+will arrive in a later time with a next phase. Following is the initial idea how
+it could work.\
+To change the configuration of Default PodNetwork, we will expose a new field in
+the Node spec object called overrideDefaultPodNetwork. This field will allow you
+to change what is the default PodNetwork on a per-Node basis. This field will be
+mutable. When kubelet is going to report the presence of the Default PodNetwork,
+it will first look at this field, and then fallback to the “default” name.\
+For the migration process, the installer will have to set that field to the new
+PodNetwork, and when the “default” PodNetwork is no longer “InUse”, it can be
+deleted and replaced with a new version. At this point the installer will have
+to clear up the Node’s overrideDefaultPodNetwork field.
+
+### PodNetwork Controller
+This KEP will introduce a new controller in KCM. Its main function will be:
+* Handle PodNetwork and PodNetworkAttachment conditions
+* Handle PodNetwork and PodNetworkAttachment finalizer to prevent deletion when
+InUse
+* Automatically create the “default” PodNetwork
+
+### Feature gate
+This feature will introduce a new feature gate to the --feature-gates argument
+and will be named **MultiNetwork**. All changes proposed in this design will be
+behind this gate.
+
+### Attaching PodNetwork to a Pod
+We will extend the Pod spec with a new field Pod.PodSpec.Networks. It will be a
+list allowing attaching PodNetworks to a Pod. This list is explicit, and only
+the listed PodNetworks will be attached to the Pod. When the Networks field is
+not specified, Cluster Default PodNetwork will be attached (and set in that field)
+to the Pod (to keep backward compatibility). Proposed changes are as follows:
+```go
+// PodSpec is a description of a pod.
+type PodSpec struct {
+[...]
+        // Networks is a list of PodNetworks that will be attached to the Pod.
+        //
+        // +kubebuilder:default=[{podNetworkName: “default”}]
+        // +optional
+        Networks []Network `json:"networks,omitempty"`
+}
+
+// Network defines what PodNetwork to attach to the Pod.
+type Network struct {
+        // PodNetworkName is name of PodNetwork to attach
+        // Only one of: [PodNetworkName, PodNetworkAttachmentName] can be set
+        //
+        // +optional
+        PodNetworkName string `json:"podNetworkName,omitempty"`
+
+        // PodNetworkAttachmentName is name of PodNetwork to attach
+        // Only one of: [PodNetworkName, PodNetworkAttachmentName] can be set
+        //
+        // +optional
+        PodNetworkAttachmentName string `json:"podNetworkAttachmentName,omitempty"`
+
+        // InterfaceName is the network interface name inside the Pod for this attachment.
+        // This field functionality is dependent on the implementation and its support
+        // for it.
+        // Examples: eth1 or net1
+        //
+        // +optional
+        InterfaceName string `json:"interfaceName,omitempty"`
```

Comment:
This is to ensure the name is unique across the list of Networks specified. Otherwise there is no control, especially if I will attach 2 PodNetworks from 2 different vendors.